embassy-nxp/src/pac_utils.rs

```diff
@@ -4,44 +4,44 @@
 /// Due to the type system of peripherals, access to the settings of a single pin is possible only
 /// by a single thread at a time. Read/Write operations on a single registers are NOT atomic. You
 /// must ensure that the GPIO registers are not accessed concurrently by multiple threads.
-pub(crate) fn gpio_reg() -> &'static lpc55_pac::gpio::RegisterBlock {
-    unsafe { &*lpc55_pac::GPIO::ptr() }
+pub(crate) fn gpio_reg() -> &'static crate::pac::gpio::RegisterBlock {
+    unsafe { &*crate::pac::GPIO::ptr() }
 }
```

Comment:
I gave both reference manuals a skim and it appears that the GPIO peripheral on the RT1010 and RT600 are different. So this whole file is probably just `lpc55_rt600` stuff. It's probably going to end up being cfg spam anyways because lpc55 exists.

Let alone that imxrt-ral has a different API structure here.

Namely the RT1010 has no PINT driver.

Really this is just a note because this is not enabled for the rt600 anyways.

Reply:
I think for this we could introduce `gpio/mod.rs` and equivalent `gpio/v1.rs`, `gpio/v2.rs`, ..., `gpio/vN.rs` for each variant whenever applicable.

Ultimately, having all families supported on the RAL would be much, much better.